_includes/locale/Locale.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+
+  declare(strict_types=1);
+
+  namespace Keyman\Site\com\keyman;
+
+  class Locale {
+    public const DEFAULT_LOCALE = 'en';
+
+    public const CROWDIN_LOCALES = array(
+      'en',
+      'es-ES',
+      'fr-FR'
+    );
+
+    // xx-YY locale as specified in crowdin %locale%
+    private static $currentLocale = Locale::DEFAULT_LOCALE;
+
+    // If set to true, warn about untranslated strings
+    private static $debugMode = false; // Keep false in production
+
+    // Associative array to hold translated strings in "current locale" and English
+    private static $langArray = [];
+    private static $langArrayEn = [];
+
+    /**
+     * Return the current locale. Fallback to 'en'
```

Comment:
Fallback should really be through a path, e.g. given `es-ES`, we should fallback first to `es`, then to `en`.

Reply:
So should setLocale() split on dashes for determining fallback.
e.g. with `es-ES` we end up with
```php
self::$currentLocales = ['es-ES', 'es', 'en'];
```

Reply:
There are algorithms for fallback already. See https://cldr.unicode.org/development/development-process/design-proposals/language-distance-data

That might be too much for us. @srl295 may have suggestions on simpler approaches.

Reply:
Moved to `calculateFallbackLocales()` where we can always reimplement in a TODO